### PR TITLE
Remove references to the generic :py:obj: role

### PR DESCRIPTION
**Pull request by stephenfin**
tl;dr: We shouldn't be using the `obj` roles from the Python domain.

The Python domain provides the `obj` role as a generic, "catch all" way to refer to any Python object of an unspecified type [1]. Unfortunately we were erroneously making use of this role in two places.

The first of these is our doc fields [2], which we use to provide information about request and response attributes, such as query parameters (e.g. `:query foo:`) or request parameters in the body (e.g. `:param bar:`). These are implemented using `TypedField` because we want to provide type information. For example:

    :query offset: offset number. default is 0
    :type offset: int

The `TypedField` class allows you to specify a `typerolename`, which determines the role that is automatically associated with a parameter type with an object. For example, consider this example for the Python domain:

    .. py:class:: User

       A user.

    .. py:function:: delete_user(user)

       Delete a user.

       :param user: The user to be deleted
       :type user: User
       :rtype: :class:`None`

Because the Python domain implements support for the `param` doc field using typerolename='class', the `User` in `:type user: User` effectively resolves to `:type user: :py:class:`User`` (*effectively* - there are some differences). Since we don't provide an `obj` role like the Python domain does, we probably don't want to use this. We certainly don't want to link to things in the Python domain, if that was even an option. As such we simply drop the `typerolename` attribute from our `TypedField` usages, thus disabling this "auto-linking" (which never would have worked anyway).

The other place we're using the `obj` role is in our definition of `object_types` as part of the `HTTPDomain` domain. This attribute is supposed to list the type of objects that the domain will index - things like classes and functions for the Python domain, or PUT and GET requests for the HTTP domain. This is implemented by a mapping of types to `ObjType` instances, which in turn take the following arguments [3]:

- *lname*: localized name of the type (do not include domain name)
- *roles*: all the roles that can refer to an object of this type
- *attrs*: object attributes -- currently only "searchprio" is known,
  which defines the object's priority in the full-text search index,
  see :meth:`Domain.get_objects()`.

We were passing two roles for each object type. The first of these mapped to the name of a directive used to document said object, e.g. an `:options:` role to refer to an `.. option::` directive. The latter was a generic `:obj:` role. This is clearly a hangover from the Python directive, where, as noted previously, this would allow one to use the `:obj:` role to refer to *any* indexed Python object rather than using object-specific roles like `:class:`. However, have no such generic role defined for the HTTP domain and it probably wouldn't make sense to have one. Therefore, we can and should simply drop this.

[1] https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#role-py-obj
[2] https://github.com/sphinx-doc/sphinx/blob/v4.1.2/sphinx/util/docfields.py#L1-L10
[3] https://github.com/sphinx-doc/sphinx/blob/v4.1.2/sphinx/domains/__init__.py#L40-L44

Closes: #53

### CODE REVIEW

**Comment by stephenfin** [comment]
Here's a sample `index.rst` that you can use to test this out:

```
=======                                             
Bug #53                                             
=======                                             

HTTP domain
-----------

.. http:get:: /posts/(int:post_id)

   Fetch an existing post

   :param post_id: post's unique id
   :type post_id: int
   :status 200: success
   :status 404: post not found

.. http:post:: /posts/(int:post_id)

   Replies a comment to the post.

   :param post_id: post's unique id
   :type post_id: int
   :form email: author email address
   :form body: comment body
   :reqheader Accept: the response content type depends on
       :mailheader:`Accept` header
   :reqheader Authorization: optional OAuth token to authenticate
   :resheader Content-Type: this depends on :mailheader:`Accept`
       header of request
   :status 302: and then redirects to :http:get:`/posts/(int:post_id)`
   :status 400: when form parameters are missing
   :status 404: post not found

Python domain
-------------

.. py:class:: User

   A user.

.. py:function:: delete_user(user)

   Delete a user.

   :param user: The user to be deleted
   :type user: User
   :rtype: None
```

**Review by ashb — APPROVED** [review]

**Comment by stephenfin** [comment]
Thanks!
